src/value.rs

```diff
@@ -170,14 +182,45 @@ pub trait ToLuaMulti<'lua> {
 
 /// Trait for types that can be created from an arbitrary number of Lua values.
 ///
-/// This is a generalization of `FromLua`, allowing an arbitrary number of Lua values to participate
-/// in the conversion. Any type that implements `FromLua` will automatically implement this trait.
+/// This is a generalization of `FromLua`, allowing an arbitrary number of Lua
+/// values to participate in the conversion. Any type that implements `FromLua`
+/// will automatically implement this trait.
+///
+/// Implementing either `from_lua_multi` or `from_counted_multi` is required as
+/// the default implementations recursively call each other.
 pub trait FromLuaMulti<'lua>: Sized {
     /// Performs the conversion.
     ///
-    /// In case `values` contains more values than needed to perform the conversion, the excess
-    /// values should be ignored. This reflects the semantics of Lua when calling a function or
-    /// assigning values. Similarly, if not enough values are given, conversions should assume that
-    /// any missing values are nil.
-    fn from_lua_multi(values: MultiValue<'lua>, lua: Context<'lua>) -> Result<Self>;
+    /// In case `values` contains more values than needed to perform the
+    /// conversion, the excess values should be ignored. This reflects the
+    /// semantics of Lua when calling a function or assigning values. Similarly,
+    /// if not enough values are given, conversions should assume that any
+    /// missing values are nil.
+    fn from_lua_multi(values: MultiValue<'lua>, lua: Context<'lua>) -> Result<Self> {
+        Self::from_counted_multi(values, lua, &mut 0)
```

Comment:
I think adding a separate method like this is probably the cleanest solution that would case least breakage in dependent code. It also makes the diff a lot more concise.

The only issue I can see is someone implementing `FromLuaMulti` without overriding either method which would cause infinite recursion at runtime, but I think it's obvious that _something_ has to be specified (and it's noted in the doc).

Reply:
The benefit of having those methods call each other is that dependents only need to implement one, depending on whether they need to handle arguments non-exhaustively.

Reply:
I think this is the best solution possible atm. Alternatives are:
- depending on nightly,
- having to manually implement both methods, or
- the first solution which requires possibly a lot of updates to existing code base.